Feat : implement delete group api client

kanta1207 opened the pull request:
## Overview

<!-- Provide a brief description of the changes introduced by this PR. -->
Implement delete group API client and action. Applied it to `DeleteGroupDialog` component

## Changes

<!-- List the changes -->
<!-- Delete this section if not needed -->

- Add Delete group API client and refactor existing code in [src/lib/api/group/client/groupApiClient.client.ts](https://github.com/nishiki-tech/nishiki-frontend/pull/282/files#diff-eb437d2e48a181934f57ecc6a8ea1a56ef9d37d44816a17587e744023772f9a2)
- Add Delete group action to and refactor existing code in [src/features/groups/lib/actions.ts](https://github.com/nishiki-tech/nishiki-frontend/pull/282/files#diff-cbbd8050cd0522718c2b39af6e9a9b344061c34b347e651528e8b7ea537936ab)
- Apply action function to [src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/GroupCardDropdownMenuTriggerButton/GroupCardDropdownMenuContent/DeleteGroupDialogTriggerButton/DeleteGroupDialogContent.tsx](https://github.com/nishiki-tech/nishiki-frontend/pull/282/files#diff-8b6be56b4700920bac07eda1aa96de22f0a02a2a2b3918aba7ffd292d0bca8b9)



## Screen Captures

<!-- Add screenshots or screen recordings of the changes -->
<!-- Necessary for UI changes -->
<!-- Delete this section if not needed -->

https://github.com/nishiki-tech/nishiki-frontend/assets/99339182/ede2d368-7629-455f-9a90-91f805f7f4a0




## Assignee Checklist:

<!-- You can tick the checkboxes if not applicable -->

- [x] The naming convention of the PR title is correct (See the comment at the top of this template)
- [x] The base branch is correct (See: [Types of Branches](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#types-of-branches))
- [x] The branch name follows the [Branch Naming Conventions](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/CONTRIBUTING.md#branch-naming-conventions)
- [x] The correct assignees and reviewers have been designated for this PR
- [x] The coding style follows the [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [x] All the related issues are associated with this PR
- [x] All criteria in the associated issues are met (please tick the checkboxes)
- [x] My changes do not generate new warnings or errors (especially in the console)

## Reviewer Checklist:

<!-- You can tick the checkboxes if not applicable -->

- [ ] Double-check the "Assignee Checklist"
- [ ] The code follows the generic best practices and our [Coding Style Guide](https://github.com/nishiki-tech/nishiki-frontend/blob/develop/docs/STYLEGUIDE.md)
- [ ] The code is well-commented and easy to understand
- [ ] The UI changes accurately reflect the provided design

kanta1207 commented on src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/GroupCardDropdownMenuTriggerButton/GroupCardDropdownMenuContent/DeleteGroupDialogTriggerButton/index.tsx:
This file has changed just because of props drilling.

kanta1207 commented on src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/GroupCardDropdownMenuTriggerButton/GroupCardDropdownMenuContent/index.tsx:
This file has changed just because of props drilling of `groupId`.

kanta1207 commented on src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/GroupCardDropdownMenuTriggerButton/index.tsx:
This file has changed just because of props drilling of `groupId`.

kanta1207 commented on src/features/groups/components/GroupCardList/GroupCard/GroupCardContent/index.tsx:
This file has changed just because of props drilling of `groupId`.

nick-y-ito reviewed (changes requested):
@kanta1207 
Thank you for your review! Overall, I have nothing to say about the implementation :)

I have left comments only for code styling. Please kindly take a look and address them.

kanta1207 commented:
@nick-y-ito 
Thank you for your review!

I addressed the points you mentioned following the coding style guide in the latest commit.
Please take a look at them whenever you have a moment 🙏

nick-y-ito commented on src/features/groups/lib/actions.ts:
```diff
@@ -32,9 +33,10 @@ export const createGroup = async (
- * @param groupId - The unique identifier of the group.
- * @param inputs - The input for the function.
+ * Function to call a API client to rename a group
+ * @param groupId - The unique identifier of a group
```
> this kind of description needs to provide more helpful information.

nick-y-ito commented on src/features/groups/lib/actions.ts:
```diff
@@ -49,10 +51,25 @@ export const renameGroup = async (
 
+/**
+ * Function to call a API client to remove a group
+ * @param groupId - The unique identifier of a group
```
> this kind of description needs to provide more helpful information.

nick-y-ito reviewed (changes requested):
@kanta1207 
Thank you for the update! I still see the issues that I pointed out. Please take a look.

kanta1207 commented:
@nick-y-ito 
Sorry for missing the review points...
I fixed them! Please kindly take a look at them anytime when you're available!

nick-y-ito reviewed (approved):
@kanta1207 
Thank you for the update! LGTM :)
